Automattic/jetpack
# CLI: JPS: Update partner-provision.sh script to use /jpphp/provision endpoint

This refactor will update the script so that it uses the same code path as hosts that would call the `/jpphp/provision` endpoint directly. 

Moving towards one flow will allow us to have less, better maintained code.

To test:

- First, you'll need a partner ID and secret. See PCYsg-emq-p2
- See the docs here for how to run the script: https://github.com/Automattic/jetpack/blob/master/docs/partners/plan-provisioning.md
- There are a few flows that we need to test:
    - Site is disconnected and does not have plan. 
        - If things work properly, you should get back a JSON object with `next_url`. Paste that URL in a browser and you should end up on a Jetpack Connection screen. After approving the prompt you should be connected and have the specified plan.
    - Site is disconnected and has a plan
        - If things work properly, you should get back a JSON object with `next_url`. Paste that URL in and connect the site. Also, the site should now have a partner plan instead of the previous plan.
    - Site is connected and does not have plan
        - If things work properly, you should get back a `next_url` that points to the site and the site should have the partner plan you provisioned.
    - Site is connected and has a plan
        - If things work properly, you should get back a `next_url` that points to the site and the site should have the upgraded partner plan
    - Site is VIP Go and we're passing in a user to auto-connect. This is a special case that we can chat about in Slack.
